language_detector/main.py

```diff
@@ -1,9 +1,53 @@
 """This is the entry point of the program."""
 
-from .languages import LANGUAGES
+try:
+    from .languages import LANGUAGES
+except:
+    from languages import LANGUAGES
 
 
 def detect_language(text, languages=LANGUAGES):
     """Returns the detected language of given text."""
-    # implement your solution here
-    pass
+    spanish = []
+    english = []
+    german = []
+    #Grant easy access to the dictionaries
+    for lang in languages:
+        if lang["name"] == "Spanish":
+            spanish = lang["common_words"]
+        elif lang["name"] == "English":
+            english = lang["common_words"]
+        elif lang["name"] == "German":
+            german = lang["common_words"]
+            
+    #Create a variable with number of occurrences per language
+    english_occurrences = 0
+    german_occurrences = 0
+    spanish_occurrences = 0
```

Comment:
Again here the languages are hardcoded. I think a preferable way to store the count would be in a list of dictionaries. So after your function has run through the block of unknown text you would end up with something along the lines of:

```python
count_dict = {
    'German' = 17,
     'English' = 4,
    'French' = 32,
    'Spanish' = 0
    }
```

You could achieve this by initialising a count dictionary using:

```python
count_dict = {}
for lang in languages:
    count_dict[lang['name']] = 0
```

This would create a key (eg 'German') with the lang['name'] for each lang in languages and it would initialise the value coupled to that key to 0.